sftp/sftp.go

```diff
@@ -0,0 +1,186 @@
+package sftp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// PasswordAuth indicates password-based authentication
+	PasswordAuth = "passwordAuth"
+	// KeyAuth indicates key-based authentication
+	KeyAuth = "keyAuth"
+)
+
+// SSHConfig represents the configuration for SSH connection
+type SSHConfig struct {
+	Host       string
+	Port       int
+	User       string
+	AuthMethod string
+	PrivateKey string
+	Password   string // Password for password-based authentication
+}
+
+// SSHClient interface abstracts the SSH client
+type SSHClient interface {
+	Dial(network, addr string, config *ssh.ClientConfig) (*ssh.Client, error)
+}
+
+// SFTPClient interface abstracts the SFTP client
+type SFTPClient interface {
+	CreateNew(client *ssh.Client) (*sftp.Client, error)
```

Comment:
better to remove this from the interface as it will contain implementation details

Reply:
So should we not test the SFTP client creation?

Reply:
Addressed